client/settings/express-checkout-settings/payment-request-button-preview.js

```diff
@@ -101,87 +149,78 @@ const PaymentRequestButtonPreview = () => {
-								} }
-								options={ {
-									paymentRequest: paymentRequest,
-									style: {
-										paymentRequestButton: {
-											type: buttonType,
-											theme: theme,
-											height: `${
-												buttonSizeToPxMap[ size ] ||
-												buttonSizeToPxMap.medium
-											}px`,
-										},
-									},
-								} }
-							/>
-						) }
-				</div>
-			) }
-			{ ! isWooPayEnabled && ! isPaymentRequestEnabled && (
-				<InlineNotice icon status="info" isDismissible={ false }>
-					{ __(
-						'To preview the express checkout buttons, ' +
-							'activate at least one express checkout.',
-						'woocommerce-payments'
-					) }
-				</InlineNotice>
-			) }
-			{ isPaymentRequestSettingsPage() &&
-				isPaymentRequestEnabled &&
-				! isLoading &&
-				! paymentRequest && (
-					<InlineNotice icon status="info" isDismissible={ false }>
-						{ __(
-							'To preview the Apple Pay and Google Pay buttons, ' +
-								'ensure your device is configured to accept Apple Pay or Google Pay, ' +
-								'and view this page using the Safari or Chrome browsers.',
-							'woocommerce-payments'
-						) }
-					</InlineNotice>
+	// No need to check `isStripeEceEnabled` since that's not what controls whether the express checkout
+	// buttons are displayed or not, that's always controlled by `isPaymentRequestEnabled`.
+	if ( ! isWooPayEnabled && ! isPaymentRequestEnabled ) {
+		return (
+			<InlineNotice icon status="info" isDismissible={ false }>
+				{ __(
+					'To preview the express checkout buttons, ' +
+						'activate at least one express checkout.',
+					'woocommerce-payments'
 				) }
-			<BrowserHelpText />
-		</>
-	);
+			</InlineNotice>
+		);
+	}
+
+	if ( isWooPayEnabled ) {
+		return (
+			<ButtonPreviewWrapper theme={ theme }>
+				<WooPayButtonPreview
+					size={ size }
+					buttonType={ buttonType }
+					theme={ theme }
+				/>
+			</ButtonPreviewWrapper>
+		);
+	} else if ( window.location.protocol !== 'https:' ) {
+		return <PreviewRequirementsNotice />;
+	} else if ( isStripeEceEnabled ) {
+		return (
+			<ButtonPreviewWrapper theme={ theme }>
+				<ExpressCheckoutPreviewComponent
+					stripe={ stripe }
+					buttonType={ buttonType }
+					theme={ theme }
+					height={
+						buttonSizeToPxMap[ size ] || buttonSizeToPxMap.medium
+					}
+				/>
+			</ButtonPreviewWrapper>
+		);
+	} else if ( isPaymentRequestEnabled && ! isLoading && paymentRequest ) {
+		<ButtonPreviewWrapper theme={ theme }>
+			<PaymentRequestButtonElement
+				key={ `${ buttonType }-${ theme }-${ size }` }
+				onClick={ ( e ) => {
+					e.preventDefault();
+				} }
+				options={ {
+					paymentRequest: paymentRequest,
+					style: {
+						paymentRequestButton: {
+							type: buttonType,
+							theme: theme,
+							height: `${
+								buttonSizeToPxMap[ size ] ||
+								buttonSizeToPxMap.medium
+							}px`,
+						},
+					},
+				} }
+			/>
+		</ButtonPreviewWrapper>;
+	}
+
+	return <PreviewRequirementsNotice />;
```

Comment:
The priority of the buttons here is:

1. WooPay button if WooPay is enabled;
2. Info notice if page is not served over HTTPS (ECE and PRBs both require HTTPS to be rendered);
3. ECE buttons;
4. PRBs; and finally
5. Info notice (no buttons could be rendered).

This is how I understood the mass of ternary statements that I refactored, and it makes sense to me to have the list be prioritized like this, but more than happy to have this double checked.